### PR TITLE
Wagtail 4.2 upgrade

### PR DESCRIPTION
**Pull request by katdom13**
- Include changes from this PR: https://github.com/nhsuk/wagtail-nhsuk-frontend/pull/243
- Added Wagtail 4.2 compatibility
- Added tests for Python 3.10 and 3.11
- Remove support for versions of Wagtail < 4.1 (Wagtail 4.1 or later now required)

Note: It is important that this PR be merged first (https://github.com/nhsuk/wagtail-nhsuk-frontend/pull/243) as it provides support for Wagtail 4.0 while this current PR removes support for versions of Wagtail older than 4.1

### CODE REVIEW

File: wagtailnhsukfrontend/settings/models.py [hunk 1]
```diff
@@ -1,24 +1,12 @@
 from django.db import models
 from modelcluster.fields import ParentalKey
 from modelcluster.models import ClusterableModel
-from wagtail import VERSION as WAGTAIL_VERSION
+from wagtail.admin.panels import FieldPanel
+from wagtail.admin.panels import FieldPanel as ImageChooserPanel
+from wagtail.admin.panels import InlinePanel, MultiFieldPanel, PageChooserPanel
 from wagtail.contrib.settings.models import BaseSetting, register_setting
```
Comment (nickmoreton) [inline wagtailnhsukfrontend/settings/models.py:7]:
```suggestion
from wagtail.contrib.settings.models import BaseSiteSetting, register_setting
```

Comment (nickmoreton) [reply wagtailnhsukfrontend/settings/models.py:7]: This will need to be update in the `HeaderSettings` & `FooterSettings` classes as well

**Review by nickmoreton — COMMENTED** [review]
I left a suggestion.

I also think the __init__.py can be empty to avoid

```
RemovedInDjango41Warning: 'wagtailnhsukfrontend.settings' defines default_app_config = 'wagtailnhsukfrontend.settings.apps.SettingsAppConfig'. Django now detects this configuration automatically. You can remove default_app_config.
```

File: wagtailnhsukfrontend/settings/models.py (hunk 1 above)
Comment (katdom13) [reply wagtailnhsukfrontend/settings/models.py]: Applied the change to `HeaderSettings` & `FooterSettings` classes.

**Comment by katdom13** [comment]
Superseded by: https://github.com/nhsuk/wagtail-nhsuk-frontend/pull/253

**Pull request closed without merging**